src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -35,6 +38,11 @@ public DeleteCommand(ContactIndex contactIndex) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+
+        if (contactIndex.equals(USER_CONTACT_INDEX)) {
+            throw new CommandException(MESSAGE_CANNOT_DELETE_OWN_PROFILE);
+        }
```

Comment:
Hmm alright, so in this case, a contact index of 0 will make it to this stage? LGTM, since this is a defensive programming practice.